src/PrestaShopBundle/Resources/config/services/core/constraint_validator.yml

```
@@ -43,6 +43,13 @@ services:
     tags:
       - { name: validator.constraint_validator }
 
+  prestashop.core.constraint_validator.discount_product_segment_validator:
+    class: 'PrestaShop\PrestaShop\Core\ConstraintValidator\DiscountProductSegmentValidator'
```

```suggestion
  PrestaShop\PrestaShop\Core\ConstraintValidator\DiscountProductSegmentValidator:
```
Prefer the FQCN notation for the new services, it's easier to use them for autowiring then